File: bj-1107.py
from collections import deque
n = int(input())
m = int(input())
broken = list(map(int, input().split()))
working = [i for i in range(10) if i not in broken]
ans = 987654321
q = deque()
q.append((100, 0))
for i in working:
    q.append((i, 1))

while(len(q) > 0):
    current = q.popleft()
    print(current)
    ans = min(ans, abs(n - current[0]))
    if(current[0] >= n): continue
    else:
        for i in working:
            q.append((current[0] * 10 + i , current[1] + 1))
print(ans)

File: bj-1107.cpp
#include <iostream>
#include <cstdio>
#include <vector>
using namespace std;

bool broken[10] = {0};
vector<int> working;
int n, m, tmp, ans;

void go(int sum, int cnt)
{
    ans = min(ans, cnt + abs(n - sum));
    if(sum >= n || sum == 0) return;
    for(auto x: working)
    {
        go(sum * 10 + x, cnt + 1);
    }
}

int main()
{
    scanf("%d %d", &n, &m);
    for(int i = 0; i < m; i++)
    {
        scanf("%d", &tmp);
        broken[tmp] = true;
    }
    for(int i = 0; i < 10; i++)
    {
        if(!broken[i])
        {
            working.push_back(i);
        }
    }
    ans = abs(n - 100);
    for(auto x: working)
    {
        go(x, 1);
    }

    printf("%d\n", ans);
    return 0;
}